src/librustc_mir/transform/const_prop.rs

```diff
@@ -405,13 +405,15 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
                 }
 
                 let arg = self.eval_operand(arg, source_info)?;
+                let is_release_mode = self.tcx.sess.overflow_checks();
```

Comment:
This is the wrong way around, if overflow checks are on, we don't need to check again

Reply:
Should be `let is_release_mode = !self.tcx.sess.overflow_checks()`?

Reply:
Or just do `let oflo_checks = self.tcx.sess.overflow_checks();` and use `!oflo_checks` in the `if` condition

Reply:
changed the conditions